### PR TITLE
added safe bool converter to metadata deserialization

### PR DESCRIPTION
**Pull request by johnataylor**
@alpaix can you take a quick look - reading a bool now should be very safe. i.e. anything that is not definitely true will be false but won't cause an exception

### CODE REVIEW

File: src/NuGet.Core/NuGet.Protocol.Core.v3/Converters/SafeBoolConverter.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                case JsonToken.Integer:
+                    return ((long)reader.Value) == 1;
+                case JsonToken.StartArray:
+                    reader.Skip();
+                    return false;
+                case JsonToken.StartObject:
+                    reader.Skip();
+                    return false;
+                default:
+                    return false;
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/Converters/SafeBoolConverter.cs:34]: I wonder if you can return a null here and let default value handler populate it.

Comment (alpaix) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/Converters/SafeBoolConverter.cs:34]: reader.Skip() here as well. Then you can unite 3 last cases.

File: src/NuGet.Core/NuGet.Protocol.Core.v3/Converters/SafeBoolConverter.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            switch (reader.TokenType)
+            {
+                case JsonToken.Null:
+                    return false;
+                case JsonToken.Boolean:
+                    return serializer.Deserialize<bool>(reader);
+                case JsonToken.String:
+                    return reader.Value.ToString().Equals("true", StringComparison.OrdinalIgnoreCase);
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/Converters/SafeBoolConverter.cs:24]: Try this instead:

``` c#
bool flag;
if (Boolean.TryParse(value.Trim(), out flag))
{
  return flag;
}

return false;
```

It is already [case-insensitive](https://msdn.microsoft.com/en-us/library/system.boolean.tryparse%28v=vs.110%29.aspx).

Comment (emgarten) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/Converters/SafeBoolConverter.cs:24]: :+1:

**Comment by emgarten** [comment]
:shipit:

**Comment by alpaix** [comment]
:testit: and :shipit:

**Pull request closed without merging**